kernel/kernel-impl/src/main/java/org/sakaiproject/db/impl/BasicSqlService.java

```diff
@@ -979,6 +979,13 @@ public boolean dbWriteBinary(String sql, Object[] fields, byte[] var, int offset
 		}
 		catch (SQLException e)
 		{
+			// On mysql unless you allow serverside prepared statements then the maximum size possible is configured
+			// by max_allowed_packet. The error codes below are:
+			// 1105 max_allowed_packet too small
+			// 1118 redo log size not at least 10 times max_allowed_packet
+			if ( "mysql".equals(m_vendor) && (e.getErrorCode() == 1105 || e.getErrorCode() == 1118) ) {
+				LOG.warn(String.format("SQL '%s' failed, consider useServerPrepStmts=true on JDBC connection.", sql), e);
```

Comment:
I don't believe the use of String.format is needed here since if it enters this block its typically going to emit this message.
JCL + SLF4J:
LOG.warn("SQL " + sql + " failed, consider useServerPrepStmts=true on JDBC connection.", e);
SLF4J only:
LOG.warn("SQL {} failed, consider useServerPrepStmts=true on JDBC connection.", sql, e);

I understand the want to maintain compatibility right now so would use the first approach as it works with both.